packages/3dom/src/api/sampler-spec.ts

```diff
@@ -0,0 +1,57 @@
+/* @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FakeModelKernel} from '../test-helpers.js';
+
+import {defineSampler, SamplerConstructor} from './sampler.js';
+import {defineThreeDOMElement} from './three-dom-element.js';
+
+const ThreeDOMElement = defineThreeDOMElement();
+
+suite('api/sampler', () => {
+  suite('defineSampler', () => {
+    test('yields a valid constructor', () => {
+      const GeneratedConstructor = defineSampler(ThreeDOMElement);
+      const instance = new GeneratedConstructor(new FakeModelKernel(), {
+        id: 0,
+      });
+
+      expect(instance).to.be.ok;
+    });
+
+    suite('the generated class', () => {
+      let kernel: FakeModelKernel;
+      let GeneratedConstructor: SamplerConstructor;
+
+
+      setup(() => {
+        kernel = new FakeModelKernel();
+        GeneratedConstructor = defineSampler(ThreeDOMElement);
```

Comment:
You have to pass it in because we need to be able to stringify the whole of its implementation when creating the worker. There are other ways to do this, but by using dependency injection we can also keep the code type safe.

The only condition is that it has to implement the `ThreeDOMElement` interface defined in `api.ts`.